### PR TITLE
build dev 1.10 pr8

### PR DESCRIPTION
**Pull request by sdavtaker**
- Preparing toolchain and dependencies

*Issue #1888 , if available:*

*Description of changes:*
Added logic to detect dependencies and setup compiler basic options. This is an early draft, still needs more dependencies to deal with other platforms. 

*Check all that applies:*
- [x] Did a review by yourself.
- [x] Added proper tests to cover this PR. (If tests are not applicable, explain.)
- [x] Checked if this PR is a breaking (APIs have been changed) change.
- [x] Checked if this PR will _not_ introduce cross-platform inconsistent behavior.
- [x] Checked if this PR would require a ReadMe/Wiki update.

Check which platforms you have built SDK on to verify the correctness of this PR.
- [x] Linux
- [ ] Windows
- [ ] Android
- [x] MacOS
- [ ] IOS
- [ ] Other Platforms

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -62,10 +67,31 @@ else ()
...
+        enable_testing()
+        # Testing dependency
+        find_package(GTest 1.11 REQUIRED )
+        include(GoogleTest) # for gtest_discover_tests()
+    endif ()
+
+    # Project dependencies http library (curl or OS), tinyxml2, cJSON, ssl (open or other), CRT
+    find_package(CURL REQUIRED)
+    find_package(ZLIB REQUIRED)
+    find_package(OpenSSL REQUIRED COMPONENTS Crypto)
```
Comment (SergeyRyabinin) [inline CMakeLists.txt:86]: What about aws-lc and s2n?
There are so much done in CRT to use aws-lc.

File: CMakeLists.txt [hunk 2]
```diff
@@ -62,10 +67,31 @@ else ()
             message(FATAL_ERROR "Unknown build type: ${CMAKE_BUILD_TYPE}")
         endif ()
     endif ()
+
+    # Options definition
     option(BUILD_TESTS "If enabled, the SDK will include tests in the build" OFF)
 
+    message(STATUS "Setting toolchain and dependencies")
+    if (BUILD_TESTS)
+        message(STATUS "Building tests")
```
Comment (graebm) [inline CMakeLists.txt:76]: unrelated: isn't [BUILD_TESTING](https://cmake.org/cmake/help/latest/command/enable_testing.html) the official cmake option?

Comment (sdavtaker) [reply CMakeLists.txt:76]: Good catch, updating.

File: CMakeLists.txt [hunk 3]
```diff
@@ -40,9 +40,14 @@ else ()
     set(CMAKE_CXX_EXTENSIONS OFF)
     set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
-    message(STATUS "Building ${PROJECT_NAME} ${CMAKE_PROJECT_VERSION}")
+    # Setting build to hide symbols in targets by default
+    set(CMAKE_CXX_VISIBILITY_PRESET hidden)
```
[the remark this answers is not in the document]
Comment (sdavtaker) [reply CMakeLists.txt:44]: It seems like this only affected C++ until version 3.12 of CMAKE, not applies also to C. 
https://cmake.org/cmake/help/latest/module/GenerateExportHeader.html?highlight=cmake_cxx_visibility_preset
I will leave it like this for now, we shouldn't expose anything C at this SDK right now, but worth to dig into this further for future use cases.

**Pull request merged**